package/MDAnalysis/analysis/align.py

```diff
@@ -373,13 +413,159 @@ def alignto(mobile, reference, select="all", mass_weighted=False,
+        self.ref_atoms = self.reference.select_atoms(*select['reference'])
+        self.mobile_atoms = self.mobile.select_atoms(*select['mobile'])
+        self.natoms = self.mobile_atoms.n_atoms
+
+        kwargs.setdefault('remarks', 'RMS fitted trajectory to reference')
+
+        if filename is None:
+            path, fn = os.path.split(frames.filename)
+            filename = os.path.join(path, prefix + fn)
+
+        if os.path.exists(filename) and not force:
+            logger.warn("{0} already exists and will NOT be overwritten; use force=True if you want this".format(filename))
+
+        self.filename = filename
+
+        self.natoms = self.mobile_atoms.n_atoms
+        self.ref_atoms, self.mobile_atoms = get_matching_atoms(self.ref_atoms,
+                            self.mobile_atoms, tol_mass=tol_mass, strict=strict)
+
+        self.writer = MDA.Writer(filename, self.natoms)
+
+        if mass_weighted:
+            # if performing a mass-weighted alignment/rmsd calculation
+            self.weights = np.asarray(ref_atoms.masses)
+        else:
+            self.weights = None
+
+        if os.path.exists(self.filename) and not force:
+            logger.warn("{0} already exists and will NOT be overwritten; use force=True if you want this".format(self.filename))
+
+        logger.info("RMS-fitting on {0:d} atoms.".format(len(self.ref_atoms)))
+
+        self.quiet = quiet
+        if self.quiet:
+            # should be part of a try ... finally to guarantee restoring the log level
+            logging.disable(logging.WARN)
+
+        self._setup_frames(frames, start,stop,step)
+
+    def _prepare(self):
+        # reference centre of mass system
+        self.ref_com = self.ref_atoms.center_of_mass()
+        self.ref_coordinates = self.ref_atoms.positions - self.ref_com
+
+        # allocate the array for selection atom coords
+        self.mobile_coordinates = self.mobile_atoms.positions.copy()
+        self.rmsd = np.zeros((self.nframes,))
+
+        # R: rotation matrix that aligns r-r_com, x~-x~com
+        #    (x~: selected coordinates, x: all coordinates)
+        # Final transformed traj coordinates: x' = (x-x~_com)*R + ref_com
+        self.rot = np.zeros(9, dtype=np.float64)  # allocate space for calculation
+        self.R = np.matrix(self.rot.reshape(3, 3))
+
+    def _single_frame(self):
+
+        self.mobile_com = self.mobile_atoms.center_of_mass().astype(np.float32)
+        self.mobile_coordinates[:] = self.mobile_atoms.positions - self.mobile_com
+
+        # Need to transpose coordinates such that the coordinate array is
+        # 3xN instead of Nx3. Also qcp requires that the dtype be float64
+        # (I think we swapped the position of ref and traj in CalcRMSDRotationalMatrix
+        # so that R acts **to the left** and can be broadcasted; we're saving
+        # one transpose. [orbeckst])
+        self.rmsd[self._ts.frame] = qcp.CalcRMSDRotationalMatrix(self.ref_coordinates.T.astype(np.float64),
```

Comment:
Already on it. alignto will return old_rmsd and min_rmsd, fit_to will
return the translated/rotated mobile_atoms, min_rmsd

On Tue, May 24, 2016 at 12:03 PM kain88-de <notifications@github.com> wrote:

> In package/MDAnalysis/analysis/align.py
> <https://github.com/MDAnalysis/mdanalysis/pull/856#discussion_r64452088>:
>
> > +        #    (x~: selected coordinates, x: all coordinates)
> > +        # Final transformed traj coordinates: x' = (x-x~_com)*R + ref_com
> > +        self.rot = np.zeros(9, dtype=np.float64)  # allocate space for calculation
> > +        self.R = np.matrix(self.rot.reshape(3, 3))
> > +
> > +    def _single_frame(self):
> > +
> > +        self.mobile_com = self.mobile_atoms.center_of_mass().astype(np.float32)
> > +        self.mobile_coordinates[:] = self.mobile_atoms.positions - self.mobile_com
> > +
> > +        # Need to transpose coordinates such that the coordinate array is
> > +        # 3xN instead of Nx3. Also qcp requires that the dtype be float64
> > +        # (I think we swapped the position of ref and traj in CalcRMSDRotationalMatrix
> > +        # so that R acts **to the left** and can be broadcasted; we're saving
> > +        # one transpose. [orbeckst])
> > +        self.rmsd[self._ts.frame] = qcp.CalcRMSDRotationalMatrix(self.ref_coordinates.T.astype(np.float64),
>
> we don't need old_rmsd for the alignment though so your comment was
> correct. But that can be fixed if you remove it from _fit_to.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/MDAnalysis/mdanalysis/pull/856/files/c41cf9540291e37191f253bfde996a7e0a6f3767#r64452088>
>
-- 
Have a wonderful day,
John J. Detlefs